File: src/gen-file-index.js
const fs = require("fs");
const path = require("path");

const basePath = path.join(__dirname, "../data"); // Adjust based on your folder

const result = {};

function walk(dir, airportKey) {
  const entries = fs.readdirSync(dir, { withFileTypes: true });

  entries.forEach(entry => {
    const fullPath = path.join(dir, entry.name);
    if (entry.isDirectory()) {
      walk(fullPath, airportKey);
    } else if (entry.name.endsWith(".json") || entry.name.endsWith(".geojson")) {
      const relative = path.relative(path.join(basePath, airportKey), fullPath).replace(/\\/g, "/");
      if (!result[airportKey]) result[airportKey] = [];
      result[airportKey].push(relative);
    }
  });
}

const airports = fs.readdirSync(basePath);
airports.forEach(airport => {
  const airportDir = path.join(basePath, airport);
  if (fs.lstatSync(airportDir).isDirectory()) {
    walk(airportDir, airport);
  }
});

fs.writeFileSync(path.join(basePath, "file-index.json"), JSON.stringify(result, null, 2));

File: src/gen-file-index.py
import os
import json
from pathlib import Path

base_path = Path(__file__).resolve().parent / "../data"
base_path = base_path.resolve()

result = {
    "tracon": {},
    "enroute": {}
}

def walk(dir_path, airport_key, station):
    for root, dirs, files in os.walk(dir_path):
        for file in files:
            if file.endswith(".json") or file.endswith(".geojson"):
                full_path = Path(root) / file
                relative = full_path.relative_to(base_path).as_posix()  # 'tracon/airport/file.geojson'
                result[station].setdefault(airport_key, []).append(relative.split(f"{station}/{airport_key}/")[1])

# Go through 'tracon' and 'enroute'
for station in ["tracon", "enroute"]:
    domain_path = base_path / station
    if not domain_path.exists():
        continue
    for airport in os.listdir(domain_path):
        airport_dir = domain_path / airport
        if airport_dir.is_dir():
            walk(airport_dir, airport, station)

# Write the result to file-index.json
with open(base_path / "file-index.json", "w", encoding="utf-8") as f:
    json.dump(result, f, indent=2)
